scripts/o3de/ExportScripts/export_source_built_project.py

```diff
@@ -87,28 +87,35 @@ def export_standalone_project(ctx: exp.O3DEScriptExportContext,
         logger.setLevel(logging.ERROR)
 
     # Calculate the tools and game build paths
-    default_base_build_path = ctx.engine_path / 'build' if engine_centric else ctx.project_path / 'build'
+    default_base_path = ctx.engine_path if engine_centric else ctx.project_path
     
     if is_installer_sdk:
-            # Check if we have any monolithic entries if building monolithic
-            monolithic_artifacts = glob.glob(f'cmake/Platform/{exp.get_platform_installer_folder_name()}/Monolithic/ConfigurationTypes_*.cmake',
-                      root_dir=ctx.engine_path)
-            has_monolithic_artifacts = len(monolithic_artifacts) > 0
-
-            if monolithic_build and not has_monolithic_artifacts:
-                logger.error("Trying to create monolithic build, but no monolithic artifacts are detected in the engine installation! Please re-run the script with '-nomonolithic' and '-config profile'.")
-                return 1
-            tools_build_path = ctx.engine_path / 'bin' / exp.get_platform_installer_folder_name(selected_platform) / tool_config / 'Default'
+        # Check if we have any monolithic entries if building monolithic
+        if monolithic_build and not exp.has_monolithic_artifacts(ctx):
+            logger.error("Trying to create monolithic build, but no monolithic artifacts are detected in the engine installation! Please re-run the script with '-nomonolithic' and '-config profile'.")
+            return
+        tools_build_path = ctx.engine_path / 'bin' / exp.get_platform_installer_folder_name(selected_platform) / tool_config / 'Default'
     
+    # Note, when resolving the relative paths used for the export process, there is a debate about 
+    # whether to use the current working directory, or to forcefully set the base path to an O3DE directory.
+    # In practice, there was noticeable surprise whenever the relative folder wasn't found in an O3DE directory,
+    # and sometimes commands are carelessly run from sensitive directories (such as C:\\Windows\\System32)
+    
+    # For this reason, all deliberately external paths must be specified using absolute paths
     if not tools_build_path:
-        tools_build_path = default_base_build_path / 'tools'
-    if not launcher_build_path or not launcher_build_path.is_absolute() or \
-        not (launcher_build_path.is_relative() and pathlib.Path.cwd() in [ctx.engine_path, ctx.project_path]):
-        launcher_build_path = default_base_build_path / 'launcher'
+        tools_build_path = default_base_path / 'build/tools'
+    elif not tools_build_path.is_absolute():
+        tools_build_path = default_base_path / tools_build_path
+
```

Comment:
A logger.debug here would be nice
```suggestion
    logger.debug(f"Tools build path set to {tools_build_path}")
```